nucypher/utilities/versioning.py

```diff
@@ -135,7 +136,7 @@ def _parse_brand(cls, data: bytes) -> bytes:
         brand = data[:cls._BRAND_SIZE]
         if brand != cls._brand():
             error = f"Incorrect brand. Expected {cls._brand()}, Got {brand}."
-            if not brand.isalpha():
+            if not re.match(rb'^[\w\d ]+', brand):
```

Comment:
We should be careful here:
1. Is `\d` redundant when `\w` is used?
2. Because of the `+` this could yield partial matches for example the first x bytes (for x < 4) are alphanumeric_or_space, but the rest are not; but what we want is all 4 bytes are alphanumeric_or_space. 

For example, the following incorrect brands would pass the check.
```python
In [97]: brand = b'MK&%'

In [98]: re.match(rb'^[\w\d ]+', brand)
Out[98]: <re.Match object; span=(0, 2), match=b'MK'>

In [99]: brand = b'M?&%'

In [100]: re.match(rb'^[\w\d ]+', brand)
Out[100]: <re.Match object; span=(0, 1), match=b'M'
```

Therefore, should the regex instead be:

```suggestion
            if not re.fullmatch(rb'^[\w ]+', brand):
```

Reply:
Side note: we should add a partially valid brand to `test_versioning_invalid_brand` analogous to the ones illustrated above.

Reply:
Both good points, added `fullmatch` and a test.